src/ert/storage/local_ensemble.py

```diff
@@ -511,10 +703,227 @@ def calculate_std_dev_for_parameter(self, parameter_group: str) -> xr.Dataset:
+                        or len(obs_names_1d) != num_obs_names
+                        or len(std_vals_1d) != num_obs_names
+                    ):
+
+                        raise IndexError(
+                            "Axis 0 misalignment, expected axis0 length to "
+                            f"correspond to observation names {num_obs_names}. Got:\n"
+                            f"len(response_vals_per_real)={len(response_vals_per_real)}\n"
+                            f"len(obs_names_1d)={len(obs_names_1d)}\n"
+                            f"len(std_vals_1d)={len(std_vals_1d)}"
+                        )
+
+                    if response_vals_per_real.shape[1] != len(
+                        reals_with_responses_mask
+                    ):
+                        raise IndexError(
+                            "Axis 1 misalignment, expected axis 1 of"
+                            f" response_vals_per_real to be the same as number of realizations"
+                            f" with responses ({len(reals_with_responses_mask)}),"
+                            f"but got response_vals_per_real.shape[1]"
+                            f"={response_vals_per_real.shape[1]}"
+                        )
+
+                    combined_np_long = np.concatenate(
+                        [
+                            key_index_1d,
+                            obs_names_1d,
+                            obs_vals_1d,
+                            std_vals_1d,
+                            response_vals_per_real,
+                        ],
+                        axis=1,
+                    )
+                    long_nps.append(combined_np_long)
+
+        if not long_nps:
+            msg = (
+                "No observation: "
+                + (", ".join(observation_keys) if observation_keys is not None else "*")
+                + " in ensemble"
+            )
+            raise KeyError(msg)
+
+        long_np = numpy.concatenate(long_nps)
+
+        return ObservationsAndResponsesData(long_np)
+
+    def _unify_datasets(
+        self,
+        groups: List[str],
+        concat_dim: Literal["realization", "realizations"],
+        delete_after: bool = True,
+    ) -> None:
+        for group in groups:
+            paths = sorted(self.mount_point.glob(f"realization-*/{group}.nc"))
+
+            if len(paths) > 0:
+                xr.combine_nested(
+                    [xr.open_dataset(p, engine="scipy") for p in paths],
+                    concat_dim=concat_dim,
+                ).to_netcdf(self._path / f"{group}.nc", engine="scipy")
+
+                if delete_after:
+                    for p in paths:
+                        os.remove(p)
+
+    def _unify_responses(self, key: Optional[str] = None) -> None:
+        gen_data_keys = {
+            k
+            for k, c in self.experiment.response_info.items()
+            if c["_ert_kind"] == "GenDataConfig"
+        }
+
+        if key == "gen_data" or key in gen_data_keys:
```

Comment:
Ideally we would like for storage to be agnostic to the type of response, and let that logic be with each parameter type, so it is possible to move this to gen_data_config? The reason for this is we would like to make it easy to register new response and parameter types without having to change storage. As long as they are datasets with `values` and a way to form the S matrix and the X vector, we dont really care about the details. One example of this is the we would like to move the GEN_DATA_RFT job from semeio, and include it in ert, but ideally as a plugin, meaning that the core ert application does not know it exists, but can still store the data. Another example could be splitting out `EXT_PARAM`, and make that a plugin parameter type.

Reply:
Should this use `response_type` instead? And you have introduced `ResponseDataInitialLayout`, should that be used as well?

Reply:
Could do it, would require adding a little bit more to get it fully "generic" but yep